util/git/creds.go

```diff
@@ -303,6 +305,18 @@ func (c SSHCreds) Environ() (io.Closer, []string, error) {
 		knownHostsFile := certutil.GetSSHKnownHostsDataPath()
 		args = append(args, "-o", "StrictHostKeyChecking=yes", "-o", fmt.Sprintf("UserKnownHostsFile=%s", knownHostsFile))
 	}
+	// Handle SSH socks5 proxy settings
+	if c.proxy != "" {
+		parsedProxyURL, err := url.Parse(c.proxy)
+		if err != nil {
+			log.Errorf("Could not parse repo URL '%s'", c.proxy)
+			return nil, nil, err
+		}
+		args = append(args, "-o", fmt.Sprintf("ProxyCommand='ncat --proxy-type socks5 --proxy %s:%s --proxy-auth '%s' %%h %%p'",
```

Comment:
curious.. what do `%%h %%p` represent?

Reply:
`%h` will get replaced by the target Git server's host and `%p` will be replaced by target Git Server's post. The additional `%` char is required to escape the formatter options

Reply:
For example, you can have something like `export GIT_SSH_COMMAND='ssh -o ProxyCommand="ncat --proxy-type socks5 --proxy 127.0.0.1:1080 github.com 22"` which has hardcoded target server `github.com` and port `22`. If we want to parameterize it, you can have something like `export GIT_SSH_COMMAND='ssh -o ProxyCommand="ncat --proxy-type socks5 --proxy 127.0.0.1:1080 %h %p"`